src/Uno.UI.Composition/Composition/Visual.skia.cs

```diff
@@ -178,7 +195,17 @@ internal virtual void Render(in DrawingSession parentSession)
 		}
 
 		using var session = BeginDrawing(in parentSession);
-		Draw(in session);
+		if (_requiresRepaint)
+		{
+			_requiresRepaint = false;
+			_recorder ??= new SKPictureRecorder();
+			var recorderSession = session with { Canvas = _recorder.BeginRecording(new SKRect(float.NegativeInfinity, float.NegativeInfinity, float.PositiveInfinity, float.PositiveInfinity)) };
+			// To debug what exactly gets repainted, replace the following line with `Draw(in session);`
+			Draw(in recorderSession);
+			_picture = _recorder.EndRecording();
+		}
+
+		session.Canvas.DrawPicture(_picture);
```

Comment:
Does the `_picture` is pixels based or only display list (a.k.a. "drawing directives")? On all platforms?

Reply:
Confirmed it's  display list that are store:

> The picture recorder is used to record drawing operations made to a [SKCanvas](https://learn.microsoft.com/en-us/dotnet/api/skiasharp.skcanvas?view=skiasharp-2.88) and stored in a [SKPicture](https://learn.microsoft.com/en-us/dotnet/api/skiasharp.skpicture?view=skiasharp-2.88).

https://learn.microsoft.com/en-us/dotnet/api/skiasharp.skpicturerecorder?view=skiasharp-2.88

Reply:
Yes. If I remember correctly, I read in some chromium design documents that SkPicture's resulting bitmap is cached as a texture after the first time it's drawn on an SKCanvas, but I'm not sure.